src/hotspot/share/runtime/os.cpp

```diff
@@ -1121,6 +1122,35 @@ void os::print_environment_variables(outputStream* st, const char** env_list) {
   }
 }
 
+void os::print_jvmti_agent_info(outputStream* st) {
+#if INCLUDE_JVMTI
+  // should return all kinds of JVMTI agents, but no xrun agents
+  const JvmtiAgentList::Iterator it =JvmtiAgentList::agents();
+  bool first_agent = true;
+  while (it.has_next()) {
+    const JvmtiAgent* agent = it.next();
+    if (agent != nullptr) {
+      if (first_agent) st->print_cr("JVMTI agents:");
+      first_agent = false;
+      const char* dyninfo = "";
+      const char* instrumentinfo = "";
+      const char* loadinfo = "not loaded";
+      const char* initinfo = "not initialized";
+      const char* optionsinfo = agent->options();
+      const char* pathinfo = agent->os_lib_path();
+      if (agent->is_dynamic()) dyninfo = "dynamic";
```

Comment:
maybe just initialize it as
```
      const char* dyninfo = agent->is_dynamic() ? "dynamic" : "";
```
(and the same for instrumentinfo/loadinfo/initinfo)